packages/hoppscotch-js-sandbox/src/utils.ts

```diff
@@ -87,3 +94,86 @@ export function setEnv(
     selected: envs.selected,
   }
 }
+
+export const throwErr = (err: string) => {
+  throw new Error(err)
+}
+
+export const unsafeCast = <T>(x: any): T => x
+
+export const getObjectKeysFromHandle =
+  (vm: QuickJSContext) => (handle: QuickJSHandle) => {
+    // TODO: Check if handle actually belongs to an object and is not null ?
+    const funcHandle = vm.unwrapResult(vm.evalCode("(x) => Object.keys(x)"))
+
+    const result = pipe(
+      vm.callFunction(funcHandle, vm.undefined, handle),
+      vm.unwrapResult,
+      vm.dump
+    ) as Array<string>
+
+    funcHandle.dispose()
+
+    return result
+  }
+
+export const unsafeEffect =
+  <T>(func: (input: T) => void): ((input: T) => T) =>
+  (input) => {
+    func(input)
+
+    return input
+  }
+
+export const disposeHandlers = (handlers: QuickJSHandle[]) => {
+  handlers.forEach((handler) => {
+    handler.dispose()
+  })
+}
+
+export const setFnHandlers = <T extends string>(
+  vm: QuickJSContext,
+  handle: QuickJSHandle,
+  handlerPairs: HandleFnPairs<T>[]
+): QuickJSHandle[] => {
+  const handlers: QuickJSHandle[] = []
+
+  handlerPairs.forEach((handlerPair) => {
+    const { func, key } = handlerPair
+    const funcHandle = vm.newFunction(key, func)
+    vm.setProp(handle, key, funcHandle)
+    handlers.push(funcHandle)
+  })
+
+  return handlers
+}
+
+export const mergeEnvs = (
+  originalEnvs: Environment["variables"],
+  newEnvs: Environment["variables"]
+) => {
+  const envs: Environment["variables"] = cloneDeep(originalEnvs)
+  const keyIndexes: Record<string, number> = {}
+
+  envs.forEach(({ key }, index) => {
+    keyIndexes[key] = index
+  })
+
+  newEnvs.forEach(({ key, value }) => {
+    const keyIndex = keyIndexes[key]
+
+    if (typeof keyIndex === "number") {
+      envs[keyIndex].value = value
+    } else {
+      envs.push({ key, value })
+    }
+  })
+
+  return envs
+}
+
+export type HandleFnImplementation = VmFunctionImplementation<QuickJSHandle>
+
+export type HandleFnPairs<T> = { key: T; func: HandleFnImplementation }
+
+export const defineHandleFn = (func: HandleFnImplementation) => func
```

Comment:
Better name is `defineVmFn` or `defineVmFunc` as handles have nothing to do with this in the current state.

Also provide documentation for functions like these!